retriever/lib/repository.py

```diff
@@ -12,80 +12,96 @@
+                                                             "scripts",
+                                                             script_name))
             if not file_exists(path_script_name):
-                download_from_repository("scripts/" + script_name,
-                                         os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name)))
+                _download_from_repository("scripts/" + script_name,
+                                          os.path.normpath(
+                                            os.path.join(SCRIPT_WRITE_PATH,
+                                                         script_name)))
 
             need_to_download = False
             try:
-                file_object, pathname, desc = imp.find_module(''.join(script_name.split('.')[:-1]), [SCRIPT_WRITE_PATH])
-                new_module = imp.load_module(script_name, file_object, pathname, desc)
+                file_object, pathname, desc = imp.find_module(
+                                        ''.join(
+                                            script_name.split('.')[:-1]),
+                                            [SCRIPT_WRITE_PATH])
+                new_module = imp.load_module(
+                                        script_name, file_object,
+                                        pathname, desc)
                 m = str(new_module.SCRIPT.version)
-                need_to_download = parse_version(str(script_version)) > parse_version(m)
+
+                if parse_version(str(script_version)) > parse_version(m):
+                    need_to_download = True
+                else:
+                    need_to_download = False
             except:
                 pass
             if need_to_download:
                 try:
-                    os.remove(os.path.normpath(os.path.join(HOME_DIR, "scripts", script_name)))
-                    download_from_repository("scripts/" + script_name,
-                                             os.path.normpath(os.path.join(SCRIPT_WRITE_PATH, script_name)))
+                    os.remove(os.path.normpath(os.path.join(HOME_DIR,
+                                                            "scripts",
+                                                            script_name)))
+                    _download_from_repository("scripts/" + script_name,
+                                              os.path.normpath(
+                                                os.path.join(SCRIPT_WRITE_PATH,
+                                                             script_name)))
                 except Exception as e:
                     print(e)
                     pass
-            update_progressbar(float(index + 1) / float(total_script_count))
+            _update_progressbar(float(index + 1) / float(total_script_count))
+
+        return True
     except:
         raise
-        return
-    print("\nThe retriever is up-to-date")
+        return False
```

Comment:
Is this `return` ever going to execute? I think the `raise` will prevent that return from ever executing.